javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -219,5 +219,100 @@ object Math {
     }
   }
 
-  // TODO The methods not available in the JavaScript Math object
+  def addExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val overflow =
+      b > 0 && a > Integer.MAX_VALUE - b || b < 0 && a < Integer.MIN_VALUE - b
+    if (!overflow) a + b
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def addExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val overflow =
+      b > 0 && a > Long.MAX_VALUE - b || b < 0 && a < Long.MIN_VALUE - b
+    if (!overflow) a + b
+    else throw new ArithmeticException("Long overflow")
```

Comment:
Thanks. I needed to add an additional check whether `rSign == 0 && a != Long.MinValue && b != Long.MinValue`, but with that it should work.